### PR TITLE
*: enable more flaky test, enable more shard_count and update bazel config

### PR DESCRIPTION
**Pull request by hawkingrei**
Signed-off-by: Weizhen Wang <wangweizhen@pingcap.com>

### What problem does this PR solve?

Issue Number: ref #35247

Problem Summary:

### What is changed and how it works?

- enable more flaky test
- enable more shard_count
- split test cases into subdirectories to speed up the build and the test.

Why are test cases speed up by being splited into subdirectories?

1、execute has too many test cases in the same folder. If we change one of the test cases, we need to rebuild all of it. It will take many times.
2、bazel limit 50 shard_count.  it can not run parallel, too many test cases run at slow speed.
3、test cases run in the shard. it is isolated by the sandbox. so it can not suffer from the other test cases. the test will run more stable.

### Check List

Tests 

- [x] Unit test
- [x] Integration test

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: distsql/BUILD.bazel [hunk 1]
```diff
@@ -6,7 +6,6 @@ go_library(
         "distsql.go",
         "request_builder.go",
         "select_result.go",
-        "stream.go",
```
Comment (tisonkun) [inline distsql/BUILD.bazel:9]: Why remove this file?

File: ddl/BUILD.bazel [hunk 2]
```diff
@@ -217,6 +217,7 @@ go_test(
         "@com_github_pingcap_failpoint//:failpoint",
         "@com_github_pingcap_kvproto//pkg/metapb",
         "@com_github_pingcap_log//:log",
+        "@com_github_stretchr_testify//assert",
```
Comment (tisonkun) [inline ddl/BUILD.bazel:220]: Perhaps we don't need this dependency?

File: distsql/BUILD.bazel (hunk 1 above)
Comment (hawkingrei) [reply distsql/BUILD.bazel:9]: ```stream.go``` have been removed.

File: ddl/BUILD.bazel (hunk 2 above)
Comment (hawkingrei) [reply ddl/BUILD.bazel:220]: I think Someone use testify/assert in the ddl test.

Comment (hawkingrei) [reply ddl/BUILD.bazel:220]: It is used in the ```ddl/cancel_test.go```

**Review by tisonkun — APPROVED** [review]

**Comment by tisonkun** [comment]
@hawkingrei `stream.go` removed without modifying the BUILD.bazel file. It seems not quite smooth from my perspective. Is it possible that we keep sync between the file tree and bazel files? Or we must defer the sync until someone trigger bazel config updates manually?

File: ddl/BUILD.bazel (hunk 2 above)
Comment (hawkingrei) [reply ddl/BUILD.bazel]: I find it is useless. so I have removed all ```testify/assert```

**Comment by hawkingrei** [comment]
> @hawkingrei `stream.go` removed without modifying the BUILD.bazel file. It seems not quite smooth from my perspective. Is it possible that we keep sync between the file tree and bazel files? Or we must defer the sync until someone trigger bazel config updates manually?

First, in the cockroach or other open source community, the bazel config is updated by the developers. so I will open it to the developers. But now developers are not similar to bazel. so it needs some time. but I hope it can be implemented as soon as possibly.
Second, I think the bazel config is generated by the tools. so we can do it with a robot. it can pass to merge as soon as possible.

**Review by Yisaer — APPROVED** [review]

**Pull request merged**